interview-questions.md

```diff
@@ -7,52 +7,64 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: In React, props allows the state of a parent class to pass data to the child classes. For example, when I was working on tic-tac-toe game I used props to pass a click-counting method to a child component.
 
   Researched answer:
 
+  Props are an instance of a state that are passed to components that serve the purpose of displaying information or letting a user interact with information. Once props are in the component, they cannot be modified. Props are public facing information whereas state is more privately held information.
+
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: I know that DOM is Document Object Model so I would surmise that a DOM event is something that happens within a DOM. For example, I have coded some apps that count the number of user clicks that happen in a div element. 
 
   Researched answer:
+  A DOM event is something along the lines of a user hovering over an element in the DOM, a page loading, or keystroke monitoring. DOM events are useful when a developer wants certain functions to be triggered by certain events or actions taken by users.
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: 
+  In object-oriented programming you design your code based on instances of parent objects. When I was learning React, a component I coded was an instance of the parent Component object. Functional programming prioritizes the use of functions to build more functions. The difference is an OOP approach is that different objects of the same class can inherit similar aspects including functions.
 
   Researched answer:
+  According to Norman Ramsey's famous Stack Overflow Answer: Functional programming is best used when all of the objects are known but may change behavior. OOP is best the other way around. New and popular language make good use of both principles offering classes and inheritance along with reusable functions. OOP principles can even be used to model many functional programming features.
```

Comment:
Okay this doesn't look like it was lifted from the stack overflow article although I may be mistaken. I appreciate your summarizing of these main points.